lib/modules/manager/asdf/extract.ts

```diff
@@ -0,0 +1,40 @@
+import { logger } from '../../../logger';
+import { isSkipComment } from '../../../util/ignore';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import type { PackageDependency, PackageFile } from '../types';
+
+const upgradeableTooling: Record<
+  string,
+  Pick<PackageDependency, 'depName' | 'datasource' | 'packageName'>
+> = {
+  nodejs: {
+    depName: 'node',
+    datasource: GithubTagsDatasource.id,
+    packageName: 'nodejs/node',
```

Comment:
add node versioning

Reply:
can you explain what you mean?

Reply:
https://github.com/renovatebot/renovate/blob/ab94b9503aaa3c28d523d9c7de6d06cf20b2d84f/lib/modules/manager/npm/extract/index.ts#L201

Reply:
thanks

Reply:
@viceice I used the nvm manager as the basis for this manager and it does not define `versioning` (https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/nvm/extract.ts) is this a defect or am I missing something in my understanding?

Reply:
nw - I worked it out, I don't have a default configured so need to apply it directly

Q. Is there a sanity set of tests I should be applying to the manager that checks the completeness of the PackageDependancy array to avoid missing things like the above?